File: week1/self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> res = new ArrayList<Integer>();
        for (int i = left; i <= right; i++) {
            if (isSelfDividing(i)) {
                res.add(i);
            }
        }
        return res;
    }
    
    public boolean isSelfDividing(int n) {
        int d = n;
        while (d > 0) {
            // Truncate next digit from n
            int c = d % 10;
            // Check whether n is evenly divisible by c
            if (c == 0 || n % c != 0) {
                return false;
            }
            d /= 10;
        }
        return true;
    }
}

File: spring2020/unit1/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        l = []
        for i in range(left, right + 1):
            if self.isSelfDividing(i):
                l.append(i)
        return l

    def isSelfDividing(self, n: int):
        num = n
        while num > 0:
            rightmost = num % 10
            if rightmost == 0 or n % rightmost != 0:
                return False
            num //= 10
        return True
